js/main.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+(function () {
+  // Test date
+  // var ADDRESS = '600, 350';
+  var PRICES = [100, 150, 200, 250, 300, 350, 400, 450];
+  var TYPES = ['palace', 'flat', 'house', 'bungalo'];
+  var ROOMS = [2, 3, 1];
+  var GUESTS = [1, 2, 3];
+  var CHECKINS = ['12:00', '13:00', '14:00'];
+  var CHECKOUTS = ['12:00', '13:00', '14:00'];
+  var FEATURES = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+  var DESCRIPTIONS = ['text1', 'text2', 'text3', 'text4', 'text5', 'text6', 'text7', 'text8'];
+  // var PHOTOS = [
+  //   'http://o0.github.io/assets/images/tokyo/hotel1.jpg',
+  //   'http://o0.github.io/assets/images/tokyo/hotel2.jpg',
+  //   'http://o0.github.io/assets/images/tokyo/hotel3.jpg'
+  // ];
+  // var LOCATIONS_X = [600, 550, 350, 200, 400, 300, 250, 450];
+  // var LOCATIONS_Y = [150, 300, 200, 180, 400, 250, 350, 450];
+
+  var mapBlock = document.querySelector('.map');
+  var mapPinsBlock = document.querySelector('.map__pins');
+  var pinTemplateCont = document.querySelector('#pin').content;
+
+  mapBlock.classList.remove('map--faded');
+
+  // Creating random array element
+  var getRandomElement = function (array) {
+    var indexElement = Math.floor(Math.random() * array.length);
+    return array[indexElement];
+  };
+  // Creating random array langth
+  var getRandomLength = function (array) {
+    return Math.round(0.5 + Math.random() * array.length);
+  };
+
+  // create a new array from an old random length
+  var createListRandomLength = function (oldList) {
+    var newList = [];
+    for (var i = 0; i < getRandomLength(oldList); i++) {
+      newList.push(oldList[i]);
+    }
+    return newList;
+  };
+
+  // add 0 to photo number
+  var getPhotoNumber = function (n) {
+    n++;
+    n = '0' + String(n);
+    return n;
+  };
+
+  // random number
+  var getRandomNumber = function (min, max) {
+    var number = min + Math.random() * (max + 1 - min);
+    return Math.floor(number);
+  };
+
+  var getOffers = function (count) {
+    var offersList = [];
+    for (var i = 0; i < count; i++) {
+      var offer = {
+        'author': {
+          'avatar': 'img/avatars/user' + getPhotoNumber(i) + '.png'
```

Comment:
```suggestion
          'avatar': 'img/avatars/user0' + getPhotoNumber(i) + '.png'
```
строку можно собрать и так

Reply:
также, посмотри на цикл и его переменную `i`, как думаешь, её можно использовать как переменную часть в названии аватарки?

Reply:
да, действительно. И функция уже не нужна